File: regexlib/javascript_test_file/regexlib_6103.js
/* 6103
 * '('{2})*([^'\r\n]*)('{2})*([^'\r\n]*)('{2})*'
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"'"*5000+"! _1SLQ_2"
 */
var REGEX = /'('{2})*([^'\r\n]*)('{2})*([^'\r\n]*)('{2})*'/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '\''.repeat(i*10000) + '! _1SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}

File: regexlib/2021-5-15/python_re_test_file/regexlib_6103.py
# 6103
# '('{2})*([^'\r\n]*)('{2})*([^'\r\n]*)('{2})*'
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:""+"'"*5000+"! _1SLQ_2"

import re
from time import perf_counter

regex = """'('{2})*([^'\r\n]*)('{2})*([^'\r\n]*)('{2})*'"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "\'" * i * 10000 + "! _1SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")